kyber/__init__.py

```diff
@@ -53,11 +56,24 @@ def deploy_app(tag, force):
 def init_app():
     """ create a new app, or sync with an existing one """
     cwd = os.path.abspath('.')
-    repo = git.Repo(cwd)
-    name = init.get_default_name(repo)
-    if name is None:
-        click.prompt("Unable to derive a name from the current git repository or directory!")
-        sys.exit(1)
+
+    try:
+        repo = git.Repo(cwd)
+        suggested_name = init.get_default_name(repo)
+    except NotGitRepository:
+        try:
+            repo = git.Repo.discover(cwd)
+        except NotGitRepository:
+            click.echo("No repository found")
+            sys.exit(1)
+        suggested_name = cwd.split('/')[-1]
+
+    if suggested_name is None:
+        click.echo("Unable to derive a name from the current git repository or directory!")
+        sys.exit(2)
```

Comment:
Is there any way to refactor this into  `init.get_default_name` so it can be unit tested more easily?